include/seqan3/range/views/as_const.hpp

```diff
@@ -0,0 +1,90 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::views::as_const.
+ */
+
+#pragma once
+
+#include <seqan3/core/type_traits/function.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+
+//!\brief Function object for seqan3::views::as_const.
+struct as_const_fn
+{
+    //!\brief Operator for returning values as values.
```

Comment:
```suggestion
    //!\brief Operator for returning rvalue as prvalue.
```

Reply:
Maybe link to the value categories of std as well.

Reply:
But it doesn't change the value category.